### PR TITLE
Issue 319

### LINKED ISSUE #319: java.lang.ArrayIndexOutOfBoundsException:
Caught while running on akka

```
error: Failed to fix /Users/ollie/dev/akka/akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala
java.lang.ArrayIndexOutOfBoundsException: -1
	at scala.meta.tokens.Tokens.apply(Tokens.scala:30)
	at scalafix.util.TokenList.prev(TokenList.scala:57)
	at scalafix.internal.patch.ImportPatchOps$.loop$1(ImportPatchOps.scala:44)
	at scalafix.internal.patch.ImportPatchOps$.fallbackToken(ImportPatchOps.scala:46)
	at scalafix.internal.patch.ImportPatchOps$.superNaiveImportPatchToTokenPatchConverter(ImportPatchOps.scala:88)
	at scalafix.patch.Patch$.semanticApply(Patch.scala:177)
```

### PR DESCRIPTION
**Pull request by ag91**
Closes #319 

Thanks to @GoldenZero

### CODE REVIEW

File: scalafix-tests/unit/src/test/scala/scalafix/tests/core/util/TokenListTest.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.scalatest.FunSuite
+import scala.meta._
+import scalafix.util.TokenList
+
+class TokenListTest extends FunSuite {
+
+  test("Prev returns the firstToken when is given the firstToken") {
+    val tokens = "".tokenize.get // this contains two tokens: beginningOfFile and endOfFile
+    val firstToken = tokens.last
```
Comment (olafurpg) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/core/util/TokenListTest.scala]: `tokens.head`?

File: scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala [hunk 2]
```diff
@@ -53,7 +53,7 @@ class TokenList(tokens: Tokens) {
 
   def prev(token: Token): Token = {
     tok2idx.get(token) match {
-      case Some(i) if tokens.length > i - 1 =>
+      case Some(i) if i - 1 > 0 =>
```
Comment (olafurpg) [inline scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala]: `i > 0`?

File: scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala [hunk 3]
```diff
@@ -53,7 +53,7 @@ class TokenList(tokens: Tokens) {
 
   def prev(token: Token): Token = {
     tok2idx.get(token) match {
-      case Some(i) if tokens.length > i - 1 =>
```
Comment (olafurpg) [inline scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala]: Nice catch! Bad copy paste 😅

File: scalafix-tests/unit/src/test/scala/scalafix/tests/core/util/TokenListTest.scala [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package scalafix.tests.core.util
+
+import org.scalatest.FunSuite
+import scala.meta._
+import scalafix.util.TokenList
+
+class TokenListTest extends FunSuite {
```
Comment (olafurpg) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/core/util/TokenListTest.scala]: Great idea 👍

File: scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala (hunk 2 above)
Comment (olafurpg) [reply scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala]: `tokens(0)` should be possible

File: scalafix-tests/unit/src/test/scala/scalafix/tests/core/util/TokenListTest.scala [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  test("Prev returns the firstToken when is given the firstToken") {
+    val tokens = "".tokenize.get // this contains two tokens: beginningOfFile and endOfFile
+    val firstToken = tokens.last
+    val tokenList = new TokenList(tokens)
+    assert(tokenList.prev(firstToken) == firstToken)
+    assert(tokenList.prev(tokens.last) == firstToken)
+  }
+
+  test("Slice gives an empty list with same token as inputs") {
+    val tokens = "val x = 2".tokenize.get
```
Comment (olafurpg) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/core/util/TokenListTest.scala]: do we need `val x = 2`? It seems like we could reuse the same `"".tokenize.get` by putting in the top of the suite before the first `test`. That would to write small and declarative tests without duplicating `new TokenList(..)` in every unit test.

**Comment by ag91** [comment]
Sorry, I rushed a bit XD

**Pull request merged**